From: Ben Hutchings <ben@decadent.org.uk>
Date: Thu, 19 May 2011 04:36:41 +0100
Subject: [PATCH 066/166] e1000e: Disable run-time PM support

Run-time PM was not mature in 2.6.32, and there is no way for the
user to override it.  So disable the operations in e1000e to avoid
the risk of regression.
---
 drivers/net/e1000e/netdev.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/e1000e/netdev.c b/drivers/net/e1000e/netdev.c
index c10c019..a4f9292 100644
--- a/drivers/net/e1000e/netdev.c
+++ b/drivers/net/e1000e/netdev.c
@@ -4793,7 +4793,7 @@ static int e1000_resume(struct device *dev)
 }
 #endif /* CONFIG_PM_SLEEP */
 
-#ifdef CONFIG_PM_RUNTIME
+#if 0 /* defined(CONFIG_PM_RUNTIME) */
 static int e1000_runtime_suspend(struct device *dev)
 {
 	struct pci_dev *pdev = to_pci_dev(dev);
@@ -4839,7 +4839,7 @@ static int e1000_runtime_resume(struct device *dev)
 	adapter->idle_check = !dev->power.runtime_auto;
 	return __e1000_resume(pdev);
 }
-#endif /* CONFIG_PM_RUNTIME */
+#endif /* 0, was CONFIG_PM_RUNTIME */
 #endif /* CONFIG_PM */
 
 static void e1000_shutdown(struct pci_dev *pdev)
@@ -5507,8 +5507,10 @@ MODULE_DEVICE_TABLE(pci, e1000_pci_tbl);
 #ifdef CONFIG_PM
 static const struct dev_pm_ops e1000_pm_ops = {
 	SET_SYSTEM_SLEEP_PM_OPS(e1000_suspend, e1000_resume)
+#if 0
 	SET_RUNTIME_PM_OPS(e1000_runtime_suspend,
 				e1000_runtime_resume, e1000_idle)
+#endif
 };
 #endif
 
-- 
1.7.5.4

